internal/fuse/dir.go

```diff
@@ -72,35 +65,48 @@ func replaceSpecialNodes(ctx context.Context, repo restic.Repository, node *rest
 
 func newDirFromSnapshot(ctx context.Context, root *Root, inode uint64, snapshot *restic.Snapshot) (*dir, error) {
 	debug.Log("new dir for snapshot %v (%v)", snapshot.ID(), snapshot.Tree)
-	tree, err := root.repo.LoadTree(ctx, *snapshot.Tree)
+	return &dir{
+		root: root,
+		node: &restic.Node{
+			AccessTime: snapshot.Time,
+			ModTime:    snapshot.Time,
+			ChangeTime: snapshot.Time,
+			Mode:       os.ModeDir | 0555,
+			Subtree:    snapshot.Tree,
+		},
+		inode: inode,
+	}, nil
+}
+
+func (d *dir) Open() error {
```

Comment:
Is this `Open` method used by any external caller or just internally by ReadDirAll and Lookup? In the latter case please change the signature to `func (d *dir) open(ctx context.Context) error`. That avoid also the problematic use of `context.TODO()`

Reply:
changed it to `open(context.Context)`